src/AutoMapper/QueryableExtensions/ProjectionExpression.cs

```diff
@@ -104,7 +105,10 @@ internal class MemberVisitor : ExpressionVisitor
         {
             protected override Expression VisitLambda<T>(Expression<T> node)
             {
-                var memberExpression = node.Body as MemberExpression;
+                MemberExpression memberExpression = 
+                    node.Body.NodeType == ExpressionType.Convert 
+                        ? (node.Body as UnaryExpression)?.Operand  as MemberExpression
+                        :  node.Body as MemberExpression;
```

Comment:
@lbargaoanu, 
thanks for more universal fix for this issue, and for merging it in!
 just created a new PR #1965 with promised integration tests, sorry for delay, it took time to figure out how to execute them first time.